packages/fxa-auth-server/lib/luaScripts/pruneSessionTokens.lua

```diff
@@ -0,0 +1,27 @@
+local function empty(t)
+  for _ in pairs(t) do
+    return false
+  end
+  return true
+end
+
+local function decode(value)
+  if not value then
+    return {}
+  end
+  return cjson.decode(value)
+end
+
+local tokens = decode(redis.call('get', KEYS[1]))
+local tokenIds = decode(ARGV[1])
```

Comment:
Reading through the [redis lua docs](https://redis.io/commands/eval#atomicity-of-scripts), I'm confused by the difference between KEYS and ARGV.

Reply:
I had to research this too. When using a sharded redis cluster KEYS gets used for routing and to ensure all KEYS present are on the same node (otherwise it's an error). ARGV get no special consideration. If we ever decide to use a sharded cluster we'll have to make sure our keys play nicely with the [key distribution model](https://redis.io/topics/cluster-spec#keys-distribution-model) (they don't currently, but could be made to)